app/models/agents/reddit_agent.rb

```diff
@@ -0,0 +1,80 @@
+module Agents
+  class RedditAgent < Agent
+    cannot_receive_events!
+
+  	default_schedule "every_1d"
+
+    description <<-MD
+      The Reddit Agent will scrape a subreddit frontpage for entries above an upvote treshold.
+    MD
+
+    event_description <<-MD
+          {
+            "title": "New Disney/Pixar Short \"Piper\"",
+            "url": "https://vimeo.com/189901272",
+            "comments_url": "https://www.reddit.com/r/videos/comments/5arlo9/new_disneypixar_short_piper/",
+            "score": 6172
+          }
+    MD
+
+    def default_options
+      {
+        'expected_update_period_in_days' => "2",
+        'subreddit' => 'bitcoin',
+        'minimum_score' => 40
+      }
+    end
+
+    def working?
+      event_created_within?(options['expected_update_period_in_days']) && !recent_error_logs?
+    end
+
+    def validate_options
+      unless %w[expected_update_period_in_days subreddit minimum_score].all? { |field| options[field].present? }
+        errors.add(:base, "All fields are required")
+      end
+    end
+
+    def check
+      response = HTTParty.get("http://reddit.com/r/#{options['subreddit']}", headers: {"User-Agent" => "huginn-crawler"})
```

Comment:
If you do `interpolated['subreddit']` then any Liquid templating in the `subreddit` option will be evaluated, which might be handy.